control-plane/agents/core/src/server.rs

```diff
@@ -66,6 +86,28 @@ pub(crate) struct CliArgs {
     #[structopt(long, short)]
     jaeger: Option<String>,
 }
+impl CliArgs {
+    fn args() -> Self {
+        CliArgs::from_args()
+    }
+}
+
+fn parse_key_value(source: &str) -> Result<KeyValue, String> {
+    match source.split_once('=') {
+        None => Err("Each element must be in the format: 'Key=Value'".to_string()),
+        Some((key, value)) => Ok(KeyValue::new(key.to_string(), value.to_string())),
+    }
+}
+fn default_tracing_tags() -> Vec<KeyValue> {
+    use git_version::git_version;
+    vec![
+        KeyValue::new(
+            "git.commit",
+            git_version!(args = ["--abbrev=12", "--always"]),
+        ),
+        KeyValue::new("crate.version", env!("CARGO_PKG_VERSION")),
+    ]
+}
```

Comment:
This function seems to be duplicated in a number of packages. Can it be made common to all packages?
Similarly for `parse_key_value` and `default_tracing_tags`.

Reply:
I've done it for the parse_key_value but we can't do it for default_tracing_tags as we don't want to share the values? I'm not sure actually

Reply:
Done for both now